server/controllers/user.js

```diff
@@ -200,4 +200,53 @@ module.exports = {
       return res.status(500).send("Internal Server Error");
     }
   },
+  async updateReviewAndUserRating(req, res) {
+    const { userId } = req.user;
+    const { reviewId, rating } = req.body;
+
+    // Updates Target Review Rating
+    try {
+      const request = await Request.findOne({ "embeddedReview._id": reviewId });
+
+      if (rating >= 1 && rating <= 5) {
+        request.embeddedReview.rating = rating;
+      } else {
+        throw new Error(
+          "Rating provided is not either in an acceptable format or is not a value within 1 to 5."
+        );
+      }
+
+      await request.save();
+      return res.status(201).send(request.toObject());
+    } catch (e) {
+      res.status(406).send(e.message);
+    }
+
+    // Updates User Rating
+    try {
+      const user = await User.findOne({ userId });
+
+      const reviews = await Review.find({ userId: userId });
+
+      if (rating >= 1 && rating <= 5) {
+        averageRating = 0;
```

Comment:
is averageRating pulled from the db or it is a newly declared variable?

Reply:
averageRating is a local variable that is being defined and set by the current user review ratings.